addons/stock/models/stock_rule.py

```diff
@@ -526,98 +513,82 @@ def _procure_orderpoint_confirm(self, use_new_cursor=False, company_id=None):
+                    orderpoints_contexts[product_context] |= orderpoint
+                else:
+                    orderpoints_contexts[product_context] = orderpoint
+
+            for orderpoint_context, orderpoints_by_context in orderpoints_contexts.items():
+                substract_quantity = orderpoints_by_context._quantity_in_progress()
+                product_quantity = orderpoints_by_context.product_id.with_context(orderpoint_context)._product_available()
+
+                for orderpoint in orderpoints_by_context:
+                    op_product_virtual = product_quantity[orderpoint.product_id.id]['virtual_available']
+                    if op_product_virtual is None:
+                        continue
+                    if float_compare(op_product_virtual, orderpoint.product_min_qty, precision_rounding=orderpoint.product_uom.rounding) <= 0:
+                        qty = max(orderpoint.product_min_qty, orderpoint.product_max_qty) - op_product_virtual
+                        remainder = orderpoint.qty_multiple > 0 and qty % orderpoint.qty_multiple or 0.0
+
+                        if float_compare(remainder, 0.0, precision_rounding=orderpoint.product_uom.rounding) > 0:
+                            qty += orderpoint.qty_multiple - remainder
+
+                        if float_compare(qty, 0.0, precision_rounding=orderpoint.product_uom.rounding) < 0:
+                            continue
+
+                        qty -= substract_quantity[orderpoint.id]
+                        qty_rounded = float_round(qty, precision_rounding=orderpoint.product_uom.rounding)
+                        if qty_rounded > 0:
+                            date = datetime.combine(orderpoint_context.get('to_date'), time.min)
+                            values = orderpoint._prepare_procurement_values(qty_rounded, date=date)
+                            procurements.append(self.env['procurement.group'].Procurement(
+                                orderpoint.product_id, qty_rounded, orderpoint.product_uom,
+                                orderpoint.location_id, orderpoint.name, orderpoint.name,
+                                orderpoint.company_id, values))
 
             try:
-                if use_new_cursor:
-                    cr.commit()
+                with self.env.cr.savepoint():
+                    self.env['procurement.group'].with_context(from_orderpoint=True).run(procurements, raise_if_exception=False)
+                    orderpoints._post_process_scheduler()
+                    orderpoints = self.env['stock.warehouse.orderpoint']
+            except ProcurementException as errors:
+                for procurement, error_msg in errors.procurement_exceptions:
+                    orderpoints_exceptions += [(procurement.values.get('orderpoint_id'), error_msg)]
+                orderpoints = [orderpoint.id for orderpoint in orderpoints if orderpoint not in [o[0] for o in orderpoints_exceptions]]
             except OperationalError:
```

Comment:
we could also add a "last safeguard" to avoid running this scheduler indefinitely if there's a bug in the code
``` python
if set(previous_orderpoint_ids.ids) == set(orderpoint_ids.ids):
    # don't try to run again the same orderpoints if the first loop failed
    # create a new activity "send a ticket to arm DOT odoo DOT com" :)
    return
````